### ISSUE TITLE
document migration for SessionSettings

### ISSUE THREAD

**Issue by DavidPerezIngeniero** [sbt/sbt#4048]
## steps

I'm migrating from SBT 0.13.16 to SBT 1.1.1

## problem

0.13.16 code, snippet from my `build.sbt`:

```
commands += Command(bd,
	(s"bd <servidor>", "establecer el servidor de base datos"),
	s"""bd <servidor>
) {
	est ⇒ ejTunel
} { (est, arg) ⇒
	val sesion = Project.session(est)
	val nue = Project.extract(est).append(Seq(baseDatosServidor in Global := arg), est)
	val nue2 = SessionSettings.reapply(sesion, nue)
	Project.extract(nue2).runTask("bd", nue2)
	nue2
}
```

SessionSettings is now private and doesn't compile in SBT 1.1.1.

## expectation

Migration guide in SBT 1.x documentation doesn't explain how to do with missing API.

**Comment by eed3si9n** [comment]
Thanks for the report.

Could we use `appendWithSession` here?
https://github.com/sbt/sbt/blob/v1.1.2/main/src/main/scala/sbt/Extracted.scala#L129-L131

**Comment by dwijnand** [comment]
TIL `SessionSettings.reapply`.

hopefully you can migrate to using `appendWithSession` which shipped in sbt 1.1.1.

**Comment by DavidPerezIngeniero** [comment]
Thanks!
